javascript/extractor/src/com/semmle/js/parser/TypeScriptASTConverter.java

```diff
@@ -826,8 +826,9 @@ private Node convertBinaryExpression(JsonObject node, SourceLocation loc) throws
 		Expression left = convertChild(node, "left");
 		Expression right = convertChild(node, "right");
 		JsonObject operatorToken = node.get("operatorToken").getAsJsonObject();
-		String operatorKind = getKind(operatorToken);
-		if ("CommaToken".equals(operatorKind)) {
+		String operator = getSourceLocation(operatorToken).getSource();
+		switch (operator) {
```

Comment:
It seems more correct to use the "kind" instead of the "source", but I suppose that is not rich enough for this task..

Reply:
There is a token kind for each operator, so we could do that but it seemed easier to misspell them, and they're so long that I was actually concerned about the performance of these string comparisons. They're names like `AsteriskAsteriskEqualsToken` and `GreaterThanGreaterThanGreaterThanEqualsToken`.